src/Microsoft.DotNet.Interactive.PowerShell.Tests/PowerShellKernelTests.cs

```diff
@@ -216,5 +217,27 @@ public async Task VerifyAllUsersProfileRuns()
                 File.Delete(_allUsersCurrentHostProfilePath);
             }
         }
+
+        [Fact]
+        public async Task PowerShell_Custom_Object_OutDisplay_Passed()
+        {
+            var props = (new Dictionary<string, object>
+                {
+                    { "prop1", "value1" },
+                    { "prop2", "value2" },
+                    { "prop3", "value3" }
+                });
+
+            var kernel = CreateKernel(Language.PowerShell);
+            var result = await kernel.SendAsync(new SubmitCode("[pscustomobject]@{ prop1 = 'value1'; prop2 = 'value2'; prop3 = 'value3' } | Out-Display"));
+            var outputs = result.KernelEvents.ToSubscribedList();
+
+            Assert.Collection(outputs,
+                              e => e.Should().BeOfType<CodeSubmissionReceived>(),
+                              e => e.Should().BeOfType<CompleteCodeSubmissionReceived>(),
+                              e => e.Should().BeOfType<DisplayedValueProduced>().Which.FormattedValues.Should().Equals(props),
```

Comment:
This is a common gotcha. `Equals` here is `object.Equals` and if it returns false, the test will still pass.

```suggestion
                              e => e.Should().BeOfType<DisplayedValueProduced>().Which.FormattedValues.Should().Be(props),
```